pkg/background/generate/generate.go

```diff
@@ -372,182 +372,215 @@ func (c *GenerateController) ApplyGeneratePolicy(log logr.Logger, policyContext
 }
 
 func getResourceInfo(object map[string]interface{}) (kind, name, namespace, apiversion string, err error) {
-	if kind, _, err = unstructured.NestedString(object, "kind"); err != nil {
-		return "", "", "", "", err
-	}
+	if _, found, err := unstructured.NestedStringSlice(object, "kinds"); !found {
+		if kind, _, err = unstructured.NestedString(object, "kind"); err != nil {
+			return "", "", "", "", err
+		}
 
-	if name, _, err = unstructured.NestedString(object, "name"); err != nil {
-		return "", "", "", "", err
-	}
+		if name, _, err = unstructured.NestedString(object, "name"); err != nil {
+			return "", "", "", "", err
+		}
 
+		if apiversion, _, err = unstructured.NestedString(object, "apiVersion"); err != nil {
+			return "", "", "", "", err
+		}
+	}
 	if namespace, _, err = unstructured.NestedString(object, "namespace"); err != nil {
 		return "", "", "", "", err
 	}
+	return
+}
 
-	if apiversion, _, err = unstructured.NestedString(object, "apiVersion"); err != nil {
-		return "", "", "", "", err
+func getResourceInfoFromRule(rule kyvernov1.Rule) (kind, name, namespace, apiversion string, err error) {
```

Comment:
It seems like this rule gets the resource info for `generate.data` and `generate.clone`, can we rename the function to reflect its purpose? Maybe `getResourceInfoForDataAndClone`?

Reply:
done